### PR TITLE
Fix issue #775

### LINKED ISSUE #775: BUG: fs.readdir fails when passing options
The following code:
```javascript
fs.readdir('/somedir', {/* options */}, callback);
```
Throws the error:
`
Error: Uncaught TypeError: callback is not a function (http://localhost:1234/tests.e31bb0bc.js:17740)
    at commonjsGlobal.onerror (mocha.947ba722.js:32655:12)
`

According to the [node documentation](https://nodejs.org/api/fs.html#fs_fs_readdir_path_options_callback), fs.readdir and fs.promises.readdir should optionally accept an options object.

### PR DESCRIPTION
**Pull request by bcheidemann**
Adds regression test and closes #775

### CODE REVIEW

File: tests/bugs/issue775.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+const util = require('../lib/test-utils.js');
+
+describe('fs.readdir fails when passing options, issue775', function () {
+  beforeEach(util.setup);
+  afterEach(util.cleanup);
+
+  it('should create a directory then call fs.readdir with options', function (done) {
```
Comment (humphd) [inline tests/bugs/issue775.js:8]: If you add `.skip` here, we could land this and enable it when a fix comes in.

**Comment by humphd** [comment]
Same deal here, this needs a rebase.

**Comment by bcheidemann** [comment]
I've done a rebase here too but will probably need to redo once #777 is merged as there will likely be conflicts.

**Comment by bcheidemann** [comment]
@humphd I've updated the readdir function to accept an options object - behaviour should be consistent with node fs.readdir except I haven't implemented the withFileTypes option yet. Will rename the pr and add WIP prefix.

**Review by humphd — CHANGES_REQUESTED** [review]
Do you want to rebase this to pick up the work in your other PRs that I just merged?

**Comment by bcheidemann** [comment]
@humphd I've done a rebase and added implementation for handling the withFileTypes option. Should be good to merge if you're happy with it =)

**Review by humphd — CHANGES_REQUESTED** [review]
Looks good, nice work.

Can you update the README in this same PR for the `options` and `withFileTypes` additions?

**Comment by bcheidemann** [comment]
> Looks good, nice work.
> 
> 
> 
> Can you update the README in this same PR for the `options` and `withFileTypes` additions?

No problem 👌

**Comment by bcheidemann** [comment]
@humphd updated the README =)

**Review by humphd — APPROVED** [review]
Nice!

**Pull request merged**